polaris-core/src/main/java/org/apache/polaris/core/storage/AccessConfig.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.storage;
+
+import java.util.Map;
+import org.apache.polaris.immutables.PolarisImmutable;
+
+@PolarisImmutable
+public interface AccessConfig {
+  Map<String, String> credentials();
+
+  Map<String, String> extraProperties();
```

Comment:
What's the need for breaking this out into separate fields? They all end up in the same map for every single caller

Reply:
The main idea is to isolate credential and non-credential properties in REST API responses. Specifically in [IcebergCatalogHandler](https://github.com/apache/polaris/pull/1504/files#diff-f1c2fbca81a1fe8224cf6c0cfee3a749b054b9a52092f6c7356b4af186bf70a2R717).

This PR does not cause any functional change, but prepares for exposing properties like `s3.endpoint` to clients later.

These properties are not always mixed together. PR #1225 brings some distinction in load table responses.